docs/public/alizer-spec.md

```diff
@@ -0,0 +1,129 @@
+# Alizer Specification
+
+## Background
+
+This document outlines the features Alizer offers and how they actually work.
+
+Currently, Alizer provides 3 detection options:
+
+- Language Detection (Language/Tools/Frameworks)
+- DevFile Detection
+- Component Detection
+
+and it is provided in 4 different formats:
+
+- Java library
+- CLI
+- NPM package
+- Go library
+
+so that it can be integrated easily in other projects.
+
+### Inner-loop scenario
+
+As a Vscode/IntelliJ plugin developer, I would like to detect the languages, tools, frameworks, services used within the
+opened user project in the IDE to provide a better user experience and tailor the functionalities I offer them.
+
+## Language Detection
+
+Language detection is based on the file `languages.yml` taken from the [GitHub's Linguist project](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml).
+Because of that, Alizer is able to recognize almost any programming languages, with a customized deeper detection of the
+ones listed below:
+
+- Java
+- Python
+- Javascript
+- Typescript
+
+Language recognition is performed by using files extensions within the source, and only the languages with a significant
+presence (>2% total files) are taken into account for further calculations.
+If Java, Python, Javascript or Typescript languages are detected, Alizer proceeds to check for the presence of tools and 
+frameworks.
+
+### Java
+
+The first step in our deeper Java detection is finding the configuration file used. If a `pom.xml` file is discovered, 
+Alizer assumes is a Maven project. The same for a `build.gradle` file which is a Gradle Project or a `build.xml` for an
+Ant project.
+
+NOTE: Maven, Gradle and Ant are saved as Tools inside the Language
```

Comment:
inside the Language: maybe inside the data structure returned by the analyze primitive ?